src/site_upload/process_flat/process_flat.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import os
+
+import awswrangler
+from shared import awswrangler_functions, decorators, enums, functions, pandas_functions, s3_manager
+
+log_level = os.environ.get("LAMBDA_LOG_LEVEL", "INFO")
+logger = logging.getLogger()
+logger.setLevel(log_level)
+
+
+def process_flat(manager: s3_manager.S3Manager):
+    new_file = False
+    if awswrangler.s3.does_object_exist(
+        f"s3://{manager.s3_bucket_name}/{manager.parquet_flat_key}"
+    ):
+        manager.move_file(
+            manager.parquet_flat_key,
+            manager.parquet_flat_key.replace(
+                enums.BucketPath.FLAT.value, enums.BucketPath.ARCHIVE.value
+            ),
+        )
+    else:
+        new_file = True
+    manager.move_file(
+        manager.s3_key,
+        manager.parquet_flat_key,
+    )
+    awswrangler_functions.generate_csv_from_parquet(
+        manager.s3_bucket_name,
+        manager.parquet_flat_key.split("/", 1)[0],
+        manager.parquet_flat_key.split("/", 1)[1],
+        f"s3://{manager.s3_bucket_name}/{manager.csv_flat_key}",
+    )
+    df = awswrangler.s3.read_parquet(f"s3://{manager.s3_bucket_name}/{manager.parquet_flat_key}")
+    column_dict = pandas_functions.get_column_datatypes(df.dtypes)
+    manager.update_local_metadata(
+        enums.ColumnTypesKeys.COLUMNS.value,
+        value=column_dict,
+        site=manager.site,
+        metadata=manager.types_metadata,
+        meta_type=enums.JsonFilename.COLUMN_TYPES.value,
+        extra_items={
+            "s3_path": f"s3://{manager.s3_bucket_name}/{manager.parquet_flat_key}",
+            "type": "flat",
+        },
+    )
+    manager.update_local_metadata(enums.TransactionKeys.LAST_DATA_UPDATE.value, site=manager.site)
+    manager.update_local_metadata(
+        enums.ColumnTypesKeys.LAST_DATA_UPDATE.value,
+        value=column_dict,
+        metadata=manager.types_metadata,
+        meta_type=enums.JsonFilename.COLUMN_TYPES.value,
+    )
```

Comment:
This is called twice with the same key and different values - can these be combined? (I might not be understanding how local metadata works)

Reply:
this is one of those 'i made this choice for backwards compatibility at the time' issues that has caused downstream boondoggles - `meta_type` determines which of the two cached metadata dicts (transaction details, column types) this is written to. So these actually are writing to completely different dicts.

If we think that needs to be revisited, I might want to do that as a followon PR.

Reply:
OK got it... Maybe this pattern would work better for my brain if the "where does this data get written" field had more prominence - like maybe as a mandatory kwarg or a positional arg, so it's easier to compare two calls and see "ah yes, this arg is different". In the calls above, the first positional arg felt like the "destination" info and the kwargs the values.

Honestly, a lot of these kwargs could use more clarity to a newcomer. Like... "write_local_metadata(value=x, metadata=y)" -- is `value=` the metadata being written and `metadata=` is metadata about the metadata? Or is `metadata=` the metadata being written and `value=` is something else.

Reply:
yeah - part of my struggle in getting onboarded back to this was remembering what all this stuff does.

This is sort of address via #70, but the names of all these things could use a second pass.